app/renderer/components/navigation/publisherToggle.js

```diff
@@ -54,6 +55,50 @@ class PublisherToggle extends React.Component {
     }
   }
 
+  setUpdateInterval () {
+    if (this.mounted === false) {
+      return
+    }
+    // The dispatch to update top sites occurs ~5 seconds after a
+    // page visit has started recording. It is important that a
+    // page data update is not triggered before this has happened.
+    const topSitesWait = ledgerUtil.milliseconds.second * 6
+    const updateWait = ledgerUtil.milliseconds.second * 3
```

Comment:
could we instead of triggering this every 3s, trigger this only after `MIN_VISIT_TIME`?

Reply:
also we have min count, so that means if you set this for 10, you need to visit this site 10 times. Maybe we should only add this timer if this visit would hit this minimum. This means that we would not add timer until you have 9 visits already logged and this is your 10 visit. Only in this case we would add timer, which would be min visit time

@ryanml wdyt?

Reply:
I think this approach would be a lot more efficient yes, especially for minimums with multiple visits. It only would not work for one scenario:

Suppose your minimums are 1 visit for 1 minute, and you've made one visit but only for 41 seconds. Visiting the site again with this approach you'd have to wait 60 seconds for the full MIN_VISIT_TIME to pass.

To get around that, maybe we could calculate the time remaining, (In the above scenario, calculate 60000 - 41000 = 19000 and set the timeout to 19000). Does that sound agreeable?

Reply:
yes let's do that